### DISCUSSION TITLE
Adding noreply.github as email

### DISCUSSION THREAD

**Discussion by OverLordGoldDragon** [community/community#42300] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

> Error adding ID+name@users.noreply.github.com: email is already in use

Can a Github-provided email address be added to my Github emails? It's to preserve commit attribution after changing username, without rebasing the commits.

**Comment by airtower-luna** [comment, marked as answer]
As I understand [the documentation](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-personal-account-settings/changing-your-github-username#your-git-commits) an ID-based noreply address should continue to work, the older name-based ones won't. I'll say the phrasing is a bit confusing though, and I can't confirm from experience. :sweat_smile:

**Reply by OverLordGoldDragon** [reply]
Hmm yes, that's what it says, then either the instructions are off or it's a bug. I'll report if no other replies. Thanks!

**Reply by OverLordGoldDragon** [reply]
Confirmed with Github Support -- all below is quoting:

<hr>

You shouldn't have to add your noreply email - the way it is constructed is meant to tolerate renaming your account. They should continue to map to contributions using the user ID part of the address. Commits made using the private noreply email are linked directly to your GitHub account, not to an email address or username.

> Git commits that were associated with your GitHub-provided noreply email address won't be attributed to your new username and won't appear in your contributions graph.

This is talking about the older format, ie: `<username>@noreply...`

> If your Git commits are associated with another email address you've added to your GitHub account, including the ID-based GitHub-provided noreply email address, they'll continue to be attributed to you

This references the newer ID-based format ie, `<user_id>+<username>@noreply...`

**Reply by TheMadSword** [reply]
> Confirmed with Github Support -- all below is quoting:
> 
> You shouldn't have to add your noreply email - the way it is constructed is meant to tolerate renaming your account. They should continue to map to contributions using the user ID part of the address. Commits made using the private noreply email are linked directly to your GitHub account, not to an email address or username.
> 
> > **Git commits that were associated with your GitHub-provided noreply email address won't be attributed to your new username and won't appear in your contributions graph.**
> 
> This is talking about the older format, ie: `<username>@noreply...`
> 
> > If your Git commits are associated with another email address you've added to your GitHub account, including the ID-based GitHub-provided noreply email address, they'll continue to be attributed to you
> 
> This references the newer ID-based format ie, `<user_id>+<username>@noreply...`

I'm confused, this goes against what Github says here : 

https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-profile/managing-contribution-settings-on-your-profile/why-are-my-contributions-not-showing-up-on-my-profile

"Commits must be made with an email address that is connected to your account on GitHub.com, **or the GitHub-provided noreply email address provided to you in your email settings, in order to appear on your contributions graph**."

The parts in bold in what you quoted seems to contradict what is in the Github doc... ?

**Comment by jstn4x420** [comment]
Try
